lib/SPIRV/LLVMToSPIRVDbgTran.cpp

```diff
@@ -1184,8 +1184,18 @@ LLVMToSPIRVDbgTran::transDbgGlobalVariable(const DIGlobalVariable *GV) {
   Ops[FlagsIdx] = transDebugFlags(GV);
 
   // Check if GV is the definition of previously declared static member
-  if (DIDerivedType *StaticMember = GV->getStaticDataMemberDeclaration())
-    Ops.push_back(transDbgEntry(StaticMember)->getId());
+  DIDerivedType *StaticMember = GV->getStaticDataMemberDeclaration();
+  Ops.push_back(StaticMember ? transDbgEntry(StaticMember)->getId()
+                             : getDebugInfoNoneId());
+
+  if (BM->getDebugInfoEIS() == SPIRVEIS_NonSemantic_Shader_DebugInfo_200)
+    // Check if GV has an associated GVE with a DIExpression
+    for (const DIGlobalVariableExpression *GVE : DIF.global_variables()) {
+      if (GVE->getVariable() == GV) {
+        Ops.push_back(transDbgExpression(GVE->getExpression())->getId());
+        break;
+      }
+    }
```

Comment:
If my understanding is correct, `<id> Variable` from `DebugGlobalVariable` serves (or can serve) the same purpose and the expression can be handled by `getGlobalVariable` function. Could you please check what the translator generates for `<id> Variable` and can it be reused later in lib/SPIRV/SPIRVToLLVMDbgTran.cpp to generate the expression?